jvm-diagnostics/src/main/java/com/palantir/jvm/diagnostics/JvmDiagnostics.java

```diff
@@ -173,4 +188,22 @@ public long getCpuTimeNanoseconds(long threadId) {
             return threadManagementBean.getThreadCpuTime(threadId);
         }
     }
+
+    private static final class HotspotCpuSharesAccessor implements CpuSharesAccessor {
+
+        private final jdk.internal.platform.Metrics metrics = jdk.internal.platform.Metrics.systemMetrics();
+
+        boolean isEnabled() {
+            return metrics.getCpuShares() != -2;
```

Comment:
From https://github.com/openjdk/jdk/blob/master/src/java.base/share/classes/jdk/internal/platform/Metrics.java#L153-L169 I think we might want to adjust this to:

> shares value, -1 if the metric is not available or
>     *         -2 if cpu shares are not supported.

```suggestion
            return metrics.getCpuShares() >= 0;
```


- https://github.com/openjdk/jdk/blob/master/src/java.base/linux/classes/jdk/internal/platform/cgroupv1/CgroupV1Subsystem.java#L245-L250
- https://github.com/openjdk/jdk/blob/master/src/java.base/linux/classes/jdk/internal/platform/cgroupv2/CgroupV2Subsystem.java#L155-L180

Reply:
Would you prefer to avoid that distinction?

Reply:
I assume you'll expose this as a gauge metric? Would you consume it as something like this?

```
JvmDiagnostics.cpuShares().ifPresent(shares -> registerGauge(() -> shares.getAsLong()))
```

Reply:
Both a gauge metric, and our startup logging (the latter is where we can distinguish between unavailable and unsupported)